File: backend/app/domain/shared/enums.py
"""Enums shared between domain entities and ORM models.

Centralized here (rather than duplicated per module) so the same value set
backs both the SQLAlchemy `Enum` columns in
`app/infrastructure/db/models.py` and any domain/application logic that
branches on them — one definition, no drift.
"""

from __future__ import annotations

from enum import StrEnum


class InventoryChangeReason(StrEnum):
    RECEIPT = "receipt"
    CONSUMPTION = "consumption"
    TRANSFER_OUT = "transfer_out"
    TRANSFER_IN = "transfer_in"
    EXPIRY_WRITEOFF = "expiry_writeoff"
    ADJUSTMENT = "adjustment"


class TransferStatus(StrEnum):
    PROPOSED = "proposed"
    APPROVED = "approved"
    IN_TRANSIT = "in_transit"
    COMPLETED = "completed"
    CANCELLED = "cancelled"


class ForecastModelType(StrEnum):
    """Which ML model produced a forecast row — not to be confused with the
    `ForecastModel` SQLAlchemy ORM class in `infrastructure/db/models.py`."""

    CHRONOS = "chronos"
    LIGHTGBM = "lightgbm"


class PurchaseOrderStatus(StrEnum):
    RECOMMENDED = "recommended"
    APPROVED = "approved"
    ORDERED = "ordered"
    RECEIVED = "received"
    CANCELLED = "cancelled"


class RecommendedBy(StrEnum):
    AI = "ai"
    MANUAL = "manual"


class AlertSeverity(StrEnum):
    INFO = "info"
    WARNING = "warning"
    CRITICAL = "critical"


class AlertType(StrEnum):
    SHORTAGE_RISK = "shortage_risk"
    EXPIRY_RISK = "expiry_risk"
    LOW_STOCK = "low_stock"
    TRANSFER_SUGGESTED = "transfer_suggested"


class HospitalType(StrEnum):
    GENERAL = "general"
    SPECIALTY = "specialty"
    TEACHING = "teaching"
    CLINIC = "clinic"

File: frontend/src/lib/types/enums.ts
/** Mirrors backend/app/domain/shared/enums.py 1:1 — one definition there,
 * one here, kept in sync by hand (see docs/architecture.md's "hand-written
 * API client" scope decision). */

export type InventoryChangeReason =
  | "receipt"
  | "consumption"
  | "transfer_out"
  | "transfer_in"
  | "expiry_writeoff"
  | "adjustment";

export type TransferStatus = "proposed" | "approved" | "in_transit" | "completed" | "cancelled";

export type ForecastModelType = "chronos" | "lightgbm";

export type PurchaseOrderStatus = "recommended" | "approved" | "ordered" | "received" | "cancelled";

export type RecommendedBy = "ai" | "manual";

export type AlertSeverity = "info" | "warning" | "critical";

export type AlertType = "shortage_risk" | "expiry_risk" | "low_stock" | "transfer_suggested";

export type HospitalType = "general" | "specialty" | "teaching" | "clinic";

export type AssignableRole = "admin" | "hospital_manager" | "pharmacist" | "viewer";
